langlib/langlib-test/src/test/java/org/ballerinalang/langlib/test/statements/foreach/ForeachErrorHandlingTests.java

```diff
@@ -48,13 +47,4 @@ public void testArrayForeachAndTrap() {
         Assert.assertEquals(returns.length, 1);
         Assert.assertEquals(((BInteger) returns[0]).intValue(), 14);
     }
-
-    @Test(expectedExceptions = BLangRuntimeException.class, expectedExceptionsMessageRegExp =
-            "error: \\{ballerina\\}TypeCastError \\{\"message\":\"incompatible types: 'error' cannot be cast to " +
-                    "'int'\"\\}\n" +
-                    "\tat foreach_error_handling:\\$lambda\\$_0\\(foreach_error_handling.bal:41\\)\n" +
-                    "\t   foreach_error_handling:\\$lambda\\$_0\\$lambda0\\$\\(foreach_error_handling.bal:40\\)")
-    public void testArrayForeachAndPanic() {
-        BValue[] returns = BRunUtil.invoke(program, "testArrayForeachAndPanic");
```

Comment:
Why are we removing this test? We can still have the same test and change the expected exception message to reflect the new error, right?

Reply:
With this new change, we cannot reproduce the expected error.

Reply:
What this seems to be [asserting is that the last line is not executed](https://github.com/ballerina-platform/ballerina-lang/blob/616e66164c3d500f125c16253d97cb845bbc24a0/langlib/langlib-test/src/test/resources/test-src/statements/foreach/foreach_error_handling.bal#L30).

With the changes in this PR, there will still be a panic due to the invalid conversion and usage of checkpanic at https://github.com/ballerina-platform/ballerina-lang/blob/9b5de12e5ab9828394d58977746ef8aab86eb464/langlib/langlib-test/src/test/resources/test-src/statements/foreach/foreach_error_handling.bal#L33.

So the last line will not be executed. We can assert that the panic is due to the invalid attempt to parse the string to an int, and thereby still assert the original behaviour, right?

Basically, we can change the `expectedExceptionsMessageRegExp` to reflect the new error and still assert the same logic that was previously asserted.

Reply:
Added the test.